models/DataObject/AbstractObject.php

```diff
@@ -407,91 +397,93 @@ protected static function typeMatch(AbstractObject $object)
 
     /**
      * @param array $objectTypes
-     * @param bool $unpublished
+     * @param bool $includingUnpublished
      *
      * @return self[]
      */
-    public function getChildren($objectTypes = [self::OBJECT_TYPE_OBJECT, self::OBJECT_TYPE_FOLDER], $unpublished = false)
+    public function getChildren(array $objectTypes = [self::OBJECT_TYPE_OBJECT, self::OBJECT_TYPE_FOLDER], $includingUnpublished = false)
     {
-        if ($this->o_children === null || $this->lastGetChildrenObjectTypes != $objectTypes) {
-            $this->lastGetChildrenObjectTypes = $objectTypes;
+        $cacheKey = $this->getListingCacheKey(func_get_args());
 
+        if (!isset($this->o_children[$cacheKey])) {
             $list = new Listing();
-            $list->setUnpublished($unpublished);
+            $list->setUnpublished($includingUnpublished);
             $list->setCondition('o_parentId = ?', $this->getId());
             $list->setOrderKey(sprintf('o_%s', $this->getChildrenSortBy()));
             $list->setOrder('asc');
             $list->setObjectTypes($objectTypes);
-            $this->o_children = $list->load();
+            $this->o_children[$cacheKey] = $list->load();
+            $this->o_hasChildren[$cacheKey] = (bool) count($this->o_children[$cacheKey]);
         }
 
-        return $this->o_children;
+        return $this->o_children[$cacheKey];
+
     }
 
     /**
      * Quick test if there are children
      *
      * @param array $objectTypes
-     * @param bool $unpublished
+     * @param bool|null $includingUnpublished
      *
      * @return bool
      */
-    public function hasChildren($objectTypes = [self::OBJECT_TYPE_OBJECT, self::OBJECT_TYPE_FOLDER], $unpublished = false)
+    public function hasChildren($objectTypes = [self::OBJECT_TYPE_OBJECT, self::OBJECT_TYPE_FOLDER], $includingUnpublished = null)
```

Comment:
array typehint missing

Reply:
I guess because it wasn't there before ... however the annotation is correct.

Reply:
but it got added on all other methods in this class :)